File: 28-Implement-strStr()/solution.java
public class Solution {
    
    public int strStr( String haystack,  String needle) 
	{
	     if(needle==null)
	        return 0;
	     if(needle.length()==0)
	        return 0;

           
	    for(int i=0;i<=haystack.length()-needle.length();i++)
	    {
	        if(ismatch(haystack,needle,i))
	        {
	            return i;
	        }
	    }
	    return -1;
	}
	public boolean ismatch(String haystack,String needle,int s)
	{
	    for(int i=0;i<needle.length();i++)
	    {
	        if(needle.charAt(i)==haystack.charAt(s))
	        {
	            s++;
	        }
	        else
	         return false;
	    }
	    return true;
	}
}

File: 28-Implement-strStr()/solution.cs
public class Solution {
    public int StrStr(string haystack, string needle) {
        int minIndex = -1;
        char[] n = needle.ToCharArray();
        char[] h = haystack.ToCharArray();
        
        if( n.Length > h.Length)
         return minIndex;
         
        if(n.Length == h.Length && n.Length == 0)//"" both are empty string
         return 0;

        int j=0;
        for(int i =0 ; i<h.Length-n.Length; i++){
            if(j==0)
              minIndex = i;
              
            else if( j == n.Length)
                return minIndex;

            if( h[i] == n[j] )
                j++;
            else
            {
                i = i-j; // return i back by j for overlaping..
                j = 0;
            }
       }
       
       if(j == n.Length)
            return minIndex;
        else
            return -1;
    }
}